pkg/test/framework/components/echo/docker/builder.go

```diff
@@ -105,6 +105,10 @@ func (b *builder) newInstances() ([]echo.Instance, error) {
 		return nil, aggregateErr
 	}
 
+	for _, inst := range instances {
```

Comment:
why this change?

Reply:
newInstances is called in parallel. This calls TrackResources, which does something like `closers = append(closers, ...)` which is not protected by any mutexs. We could instead add a mutex but I felt this was easier

Reply:
Ah good catch. TrackResources should probably be protected, no?

FYI @ozevren

Reply:
Yes, it should be protected. Thanks for fixing this.

Reply:
@ozevren @nmittler  Do you mean we should add mutex to TrackResource or is this change sufficient?

Reply:
Go ahead and add the mutex ... and then you can revert this part of the change.

Reply:
@nmittler I added a mutex